lib/aws-iot-rpi-fleet-provisioning-stack.ts

```diff
@@ -146,42 +179,76 @@ export class AwsIotRpiFleetProvisioningStack extends cdk.Stack {
       }
     });
 
-    // The bucket where the codebuild artifacts and sources are stored
-    // This is where the modified raspbian image generated by codebuild is stored
-    const provisioningArtifactsBucket = new s3.Bucket(this, 'ProvisioningArtifactsBucket', {
+    // The bucket where the configured rpi-image-builder used as a source of the pipeline is stored
+    const rpiImageBuilderSourceBucket = new s3.Bucket(this, 'rpiImageBuilderSourceBucket', {
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      versioned: true, // CodePipeline requires a versioned source for source stages
+    });
+
+    // The bucket where the custom raspbian image created by the pipeline is stored
+    const rpiImageOutputBucket = new s3.Bucket(this, 'RpiImageOutputBucket', {
       encryption: s3.BucketEncryption.S3_MANAGED,
       blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
     });
 
-    // Codebuild project that generates the raspbian image
-    const createRpiImageProject = new codebuild.Project(this, 'CreateRPIImage', {
+    // Lambda that configures the rpi-image-builder and stores it in the pipeline source bucket
+    const configureRpiImageBuilderFunction = new lambda.Function(this, 'ConfigureRpiImageBuilderFunction', {
+      runtime: lambda.Runtime.PYTHON_3_7,
+      handler: 'app.on_event',
+      code: lambda.Code.fromAsset(path.join(__dirname, './lambda/configure_rpi_image_builder')),
+      timeout: cdk.Duration.seconds(60),
+      initialPolicy: [
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            'iot:CreateKeysAndCertificate',
+            'iot:AttachPolicy',
+            'iot:DescribeEndpoint',
+          ],
+          resources: ['*'],
```

Comment:
un-scoped resource access?

Reply:
I don't see how the scope of those three actions could be restricted in this context

Reply:
yes, now I see the context - you're right